### PR TITLE
[SPARK-19471][SQL]AggregationIterator does not initialize the generated result projection before using it

### PR DESCRIPTION
**Pull request by DonnyZone**
## What changes were proposed in this pull request?

This is a follow-up PR that moves the test case in PR-18920 (https://github.com/apache/spark/pull/18920) to DataFrameAggregateSuit.

## How was this patch tested?
unit test

### CODE REVIEW

**Comment by DonnyZone** [comment]
@gatorsmile

**Comment by gatorsmile** [comment]
ok to test

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 1]
```diff
@@ -17,13 +17,18 @@
...
+import org.apache.spark.sql.execution.WholeStageCodegenExec
+import org.apache.spark.sql.execution.aggregate.{HashAggregateExec, ObjectHashAggregateExec, SortAggregateExec}
 import org.apache.spark.sql.expressions.Window
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.test.SharedSQLContext
 import org.apache.spark.sql.test.SQLTestData.DecimalData
 import org.apache.spark.sql.types.{Decimal, DecimalType}
 
+
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala]: Please remove this line. Thanks!

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 2]
```diff
@@ -573,4 +578,47 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
       spark.sql("SELECT 3 AS c, 4 AS d, SUM(b) FROM testData2 GROUP BY c, d"),
       Seq(Row(3, 4, 9)))
   }
+
+  private def assertNoExceptions(c: Column): Unit = {
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala]: Could you move it to line 566,  just before `test("SPARK-21580 ints in aggregation expressions are taken as group-by ordinal.")`? Thanks!

**Comment by gatorsmile** [comment]
LGTM except two minor comments.

**Comment by DonnyZone** [comment]
Is Jenkin unstable?

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merging to master

**Pull request closed without merging**